### PR TITLE
[MNT] updates for release 

### PR DESCRIPTION
**Pull request by SUKI-O**
#### Reference Issues/PRs

#292 

#### What does this implement/fix? Explain your changes.
Update `pyproject.toml` for releasing `treeple`
Update Licence

### CODE REVIEW

File: CITATION.cff
Comment (adam2392) [inline CITATION.cff:1]: Perhaps you should add yourself to this file?

**Review by adam2392 — COMMENTED** [review]
LGTM. What is the process you will take to make a release now? Is it clear from the developer documentation?

If not, perhaps we can simultaneously make that crystal clear, so anyone (you and me and anyone else) can do the release process easily in the future.

**Comment by adam2392** [comment]
FYI you released a version onto pypi that is not installable from non-macm1-non-python3.9 environments: https://pypi.org/project/treeple/#files.

Can you re-release v0.9.1?

If you take a look at the previous versions, there are many wheels. Lmk if there are developer documentation instructions that are unclear, or need patching. You can use testpypi to test it out. Feel free to write out a step-by-step process here if you want me to double check.

**Review by adam2392 — APPROVED** [review]

**Comment by adam2392** [comment]
In addition, you will want to:

3. Update https://github.com/neurodata/treeple/blob/main/doc/whats_new.rst 
4. Update https://github.com/neurodata/treeple/blob/main/doc/whats_new/v0.9.rst release changelog file. You can just do things like remove "In development" and add any quick summary notes that are important that we changed.
5. Add a new whats_new file for the next version.
6. Rebuild the documentation locally
7. 1. make a tag on GH for v0.9.x and freeze what you have rn according to that tag
2. make a release and upload the wheels https://github.com/neurodata/treeple/releases (this should theoretically trigger the release workflow to release on PyPi, but you already released, so I think it'll just won't do it). This provides a place for ppl to download that isn't via pypi.
8. In `gh-pages`, upload the documentation under the new version folder name (e.g. `v0.9`)
9. update the `_versions.json` in https://github.com/neurodata/treeple/blob/main/doc/_static/versions.json
10. Finally, is https://github.com/neurodata/treeple/blob/main/DEVELOPING.md#making-a-release reflective of what you have done?

`dev/` docs will take care of themselves and get uploaded via circleCI. Lmk if that makes sense and is reflected in #299 ?

**Comment by adam2392** [comment]
@SUKI-O can you address these final points for releasing v0.9.x, so that development of v0.10 can begin?

**Comment by SUKI-O** [comment]
@adam2392 can you review the changes, redux of release etc.. Sorry it took so long. LMK if I need to fix smtg else.

**Review by adam2392 — COMMENTED** [review]
Looking good!

I think this file seems like it's extra so can be removed doc/sg_execution_times.rst

Were you able to update the gh-pages with relevant documentation?

https://docs.neurodata.io/treeple/v0.9/install.html for some reason still shows the dev0 header. One thing you can do if the docs already built in there is just ctrl+f the extra portion and delete it. 0.9.dev0 -> 0.9 and update the docs in gh-pages manually. Let me know if that makes sense. 

Other than these two minor issues, it LGTM! Now you know how to deal with 99% of the OSS developer stuff out there :)

**Comment by adam2392** [comment]
> Looking good!
> 
> I think this file seems like it's extra so can be removed doc/sg_execution_times.rst
> 
> Were you able to update the gh-pages with relevant documentation?
> 
> https://docs.neurodata.io/treeple/v0.9/install.html for some reason still shows the dev0 header. One thing you can do if the docs already built in there is just ctrl+f the extra portion and delete it. 0.9.dev0 -> 0.9 and update the docs in gh-pages manually. Let me know if that makes sense.
> 
> Other than these two minor issues, it LGTM! Now you know how to deal with 99% of the OSS developer stuff out there :)

I have fixed both these issues by pushing to gh-pages and this PR branch, since it's minor.  

I  will merge this in now, and then create a new branch on GH called `v0.9.x` indicating a state for our stuff to live on in v0.9.x. Just as a FYI, this helps in backtracking any crazy bugs we need a quick fix for. We would just push/PR directly to the branch `v0.9.x`, and do a new release v0.9.2.

**Review by adam2392 — APPROVED** [review]

**Pull request merged**